AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -427,6 +427,12 @@ - (void)dealloc
 
   _subnodes = nil;
 
+#if YOGA
+  if (_yogaNode != NULL) {
```

Comment:
For the three changes in ASDisplayNode.mm, I couldn't figure out a reasonable way to move them out. However, I thought pretty carefully about each of these, and don't think they will create any complexity problems during refactoring.  The hierarchyState integration is the biggest, but it is very simple code that follows a pattern of the other if() cases